### ISSUE TITLE
MJML server received too many data error firing and not sure why

### ISSUE THREAD

**Issue by benjamindell** [liminspace/django-mjml#63]
Hi. I'm not sure if this is due to me running python3 and the latest Django but when i try and send an email the following lines are triggered in tcpserver.js

```
} else if (total_data.length > data_size) {
            result = 'MJML server received too many data';
            conn.write('1');
        } else {
```

I'm guessing the check here is simply to see if data sizes match? I was racking my brains as to how to resolve this but couldnt come up with anything. So i have just commented out that condition and the emails are now sending. 

So firstly - have I created a bigger issue by letting this error go unchecked? And secondly, do you have any ideas how I should fix this?

Thanks!

**Comment by liminspace** [comment]
@benjamindell do you use latest version of django-mjml? This checking is important because tcp-server can receive data by several portions. It works well in Python 3.6.
Tell me about your environment: OS, python version, django-mjml version, how you run tcpserver, email's size you try to send, encoding your template file etc.

**Comment by benjamindell** [comment]
Hey, sorry for the delay in replying. Just checking versions etc now.

So we have django-mjml 0.5.3 - is that the latest version?

In terms of our environment:

OS: Ubuntu 18.04.1 LTS
Python: 3.6.6
django-mjml: 0.5.3
mjml-core: 4.2.0-beta.3
mjml-cli: 4.2.0-beta.3

We run tcpserver via a supervisor conf - using the exact same structure and params as the example one you provide in the readme. The emails we send might be slightly larger than usual, due to images in there etc, but nothing too crazy.

Is there anything else you need me to check at this stage?

Thanks 🙌

**Comment by liminspace** [comment]
I don't have any idea why you get this problem, I am not at my work laptop these weeks and can't check it. 
Try to check that your templates is utf-8 and write here the size of data which is sent to tcpserver.
Also try to test small emails.
If you send images inside email check if you encode them into base64.

**Comment by benjamindell** [comment]
Cool thanks Igor. I'll check it out.

**Comment by liminspace** [comment]
I closed this issue. If you still have some problems, you can reopen it.

**Comment by danihodovic** [comment]
I'm having the same issue. 

>  If you send images inside email check if you encode them into base64.

Why do images need to be base64 encoded?

**Comment by danihodovic** [comment]
An email with the characters 🍭🍭 caused this error for me.

**Comment by liminspace** [comment]
Thanks, I'll try to fix this issue at this week.

**Comment by liminspace** [comment]
The problem has been fixed in 0.8.0 version.
